Pocket/pocket-ios
# IN-232: Add Tags Screen - User Can Select From Existing Tags

## Summary
Add Tags Screen: user can select from list of existing tags

## References 
IN-232

## Implementation Details
* Added new section in SwiftUIView called `OtherTagsView` which shows list of tags not included in the input list. 
* Added new requests methods `fetchTagsWithNoSavedItems` and `fetchTags(excluding tags: [String])` to figure out which tags to delete and which tags to show in other tags view respectively. 
* For adding tags, we had to include 2 GraphQL mutations `UpdateSavedItemRemoveTagsMutation` and `ReplaceSavedItemTagsMutation` (see ticket for slack link with context). To allow the operation to take in both mutations in a clean way, we created AnyMutation as there was a generic constraint using GraphQLMutation.

## Test Steps
- [ ] Given a user is on the Add Tags screen, 
and the user views the all tags section, 
then the user should be able to see existing tags (unless included in the text field).

- [ ] Given a user is on the Add Tags screen, 
and the user taps on one of the existing tags, 
then the user should see it appear in the text field.

## PR Checklist:

## Screenshots
https://user-images.githubusercontent.com/6743397/187229041-a0638b44-e0eb-4f62-9298-818d4e35645b.mp4
